### PR TITLE
#575 request req add lines 2

### PR DESCRIPTION
**Pull request by joshxg**
#575 

- This became a little messy as updates came into the server while I was working on this task
- Decided to just commit what I had for now!

### CODE REVIEW

File: packages/common/src/types/schema.ts [hunk 1]
```diff
@@ -729,9 +729,9 @@ export type InsertRequestRequisitionInput = {
   comment?: InputMaybe<Scalars['String']>;
   id: Scalars['String'];
   maxMonthsOfStock: Scalars['Float'];
+  minMonthsOfStock: Scalars['Float'];
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:732]: new api change

File: packages/common/src/types/schema.ts [hunk 2]
```diff
@@ -1098,9 +1098,9 @@ export type ItemSortInput = {
 
 export type ItemStatsNode = {
   __typename: 'ItemStatsNode';
+  availableMonthsOfStockOnHand: Scalars['Float'];
+  availableStockOnHand: Scalars['Int'];
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1102]: new api change

File: packages/common/src/types/schema.ts [hunk 3]
```diff
@@ -1940,6 +1935,11 @@ export type RequisitionLineNode = {
   outboundShipmentLines: InvoiceLineConnector;
   /** Quantity requested */
   requestedQuantity: Scalars['Int'];
+  /**
+   * Calculated quantity
+   * When months_of_stock < requisition.min_months_of_stock, calculated = average_monthy_consumption * requisition.max_months_of_stock - months_of_stock
+   */
+  suggestedQuantity: Scalars['Int'];
```
Comment (joshxg) [inline packages/common/src/types/schema.ts:1942]: new api change

File: packages/invoices/src/InboundShipment/api/operations.graphql [hunk 4]
```diff
@@ -244,8 +244,9 @@ query invoice($id: String!, $storeId: String!) {
                 isVisible
                 unitName
                 stats {
-                  stockOnHand
-                  monthsOfStock
+                  __typename
+                  availableStockOnHand
+                  availableMonthsOfStockOnHand
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/api/operations.graphql:249]: new api change

File: packages/invoices/src/OutboundShipment/DetailView/modals/hooks/useNextItem.test.tsx [hunk 5]
```diff
@@ -51,8 +51,8 @@ const getLines = (): InvoiceLineNode[] => [
       name: '',
       stats: {
         __typename: 'ItemStatsNode',
-        stockOnHand: 0,
-        monthsOfStock: 0,
+        availableStockOnHand: 0,
+        availableMonthsOfStockOnHand: 0,
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/DetailView/modals/hooks/useNextItem.test.tsx:55]: new api changes

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx [hunk 6]
```diff
@@ -1,12 +1,31 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
```
Comment (joshxg) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx:1]: this is a basic implementation with the building blocks in place that I was working on before API changes came through.

After the changes came through, the demo server stopped working and I needed to update queries. Then when I switched to use a local version, other changes were needed.. so I got confused what was going on 😁

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx [hunk 7]
```diff
@@ -1,12 +1,31 @@
...
+import { RequestLineEditForm } from './RequestLineEditForm';
+import {
+  useSaveRequestLines,
+  useRequestRequisitionFields,
+  useRequestRequisitionLines,
+  useIsRequestRequisitionDisabled,
+  RequestRequisitionLineFragment,
+} from '../../api';
+
+export type DraftRequestRequisitionLine = Omit<
```
Comment (joshxg) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx:22]: Just like the other modals, creating a draft line which can be saved later. This is simplified in that there is just one line per item

File: packages/requisitions/src/RequestRequisition/DetailView/columns.ts [hunk 8]
```diff
@@ -28,8 +28,8 @@ export const useRequestRequisitionColumns = ({
         'previousStockOnHand',
         {
           accessor: ({ rowData }) =>
-            `${rowData.itemStats.stockOnHand} (${Math.floor(
-              (rowData.itemStats.stockOnHand ?? 0) /
+            `${rowData.itemStats.availableStockOnHand} (${Math.floor(
```
Comment (joshxg) [inline packages/requisitions/src/RequestRequisition/DetailView/columns.ts:31]: new api change

File: packages/requisitions/src/RequestRequisition/api/operations.graphql [hunk 9]
```diff
@@ -27,11 +27,12 @@ fragment RequestRequisitionLine on RequisitionLineNode {
   id
   itemId
   requestedQuantity
-  calculatedQuantity
+  suggestedQuantity
```
Comment (joshxg) [inline packages/requisitions/src/RequestRequisition/api/operations.graphql:30]: new api change

File: packages/requisitions/src/ResponseRequisition/api/operations.generated.ts [hunk 10]
```diff
@@ -41,11 +41,12 @@ export const ResponseRequisitionLineFragmentDoc = gql`
   id
   itemId
   requestedQuantity
-  calculatedQuantity
+  suggestedQuantity
```
Comment (joshxg) [inline packages/requisitions/src/ResponseRequisition/api/operations.generated.ts:44]: new api change

File: packages/requisitions/src/ResponseRequisition/api/operations.graphql [hunk 11]
```diff
@@ -14,11 +14,12 @@ fragment ResponseRequisitionLine on RequisitionLineNode {
   id
   itemId
   requestedQuantity
-  calculatedQuantity
+  suggestedQuantity
```
Comment (joshxg) [inline packages/requisitions/src/ResponseRequisition/api/operations.graphql:17]: new api change

File: packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.tsx [hunk 12]
```diff
@@ -25,9 +25,9 @@ const toItem = (fragment: ItemRowFragment | null): Item | null => {
     availableQuantity: 0,
     stats: {
       __typename: 'ItemStatsNode',
-      monthsOfStock: 0,
+      availableMonthsOfStockOnHand: 0,
```
Comment (joshxg) [inline packages/system/src/Item/Components/ItemSearchInput/ItemSearchInput.tsx:28]: new api change

File: packages/system/src/Item/utils.ts [hunk 13]
```diff
@@ -76,9 +76,9 @@ export const toItem = (line: ItemLike): Item => ({
   availableQuantity: 0,
   stats: {
     __typename: 'ItemStatsNode',
-    monthsOfStock: 0,
+    availableMonthsOfStockOnHand: 0,
```
Comment (joshxg) [inline packages/system/src/Item/utils.ts:79]: new api change

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx [hunk 14]
```diff
@@ -31,12 +100,35 @@ export const RequestLineEdit = ({
           : t('heading.edit-item')
       }
       cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
-      nextButton={<DialogButton variant="next" onClick={() => {}} />}
-      okButton={<DialogButton variant="ok" onClick={onClose} />}
+      nextButton={<DialogButton variant="next" onClick={save} />}
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEdit.tsx:103]: I would just disable for now.. less confusing to me 🤷

File: packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx [hunk 15]
```diff
@@ -1 +1,30 @@
...
+}
+
+export const RequestLineEditForm = ({
+  item,
+  disabled,
+  onChangeItem,
+}: RequestLineEditFormProps) => {
+  const { lines } = useRequestRequisitionLines();
+  return (
+    <ItemSearchInput
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/DetailView/RequestLineEdit/RequestLineEditForm.tsx:18]: oh. unknown type `ConnectorError` etc.
I think merge this PR and then do the API changes

File: packages/requisitions/src/RequestRequisition/api/api.ts [hunk 16]
```diff
@@ -15,20 +16,32 @@ import {
...
+    requisitionId: line.requisitionId,
+    requestedQuantity: line.requestedQuantity,
+  }),
+  toUpdateLineInput: (line: DraftRequestRequisitionLine) => ({
+    id: line.id,
+    requestedQuantity: line.requestedQuantity,
+  }),
+  toInsertInput: (
+    requisition: Partial<RequestRequisitionFragment> & { id: string }
+  ): UpdateRequestRequisitionInput => {
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/api/api.ts:32]:
```suggestion
  ): InsertRequestRequisitionInput => {
```
??

File: packages/requisitions/src/RequestRequisition/api/api.ts [hunk 17]
```diff
@@ -15,20 +16,32 @@ import {
...
+    return {
+      id: requisition.id,
+      // otherPartyId: requisition.otherParty?.id,
+      comment: requisition.comment,
+      theirReference: requisition.theirReference,
+      colour: requisition.colour,
+      status:
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/api/api.ts:39]: not part of the insert input. Should this be `toUpdateInput`?

File: packages/requisitions/src/RequestRequisition/api/api.ts [hunk 18]
```diff
@@ -79,13 +92,42 @@ export const RequestRequisitionQueries = {
...
+
+      throw new Error('Unable to update requisition');
+    },
   update:
     (api: RequestRequisitionApi, storeId: string) =>
     async (
       patch: Partial<RequestRequisitionFragment> & { id: string }
     ): Promise<{ __typename: 'RequisitionNode'; id: string }> => {
-      const input = requisitionToInput(patch);
+      const input = requisitionParser.toInsertInput(patch);
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/api/api.ts:130]:
```suggestion
      const input = requisitionParser.toUpdateInput(patch);
```
🤔

File: packages/requisitions/src/RequestRequisition/api/hooks.ts [hunk 19]
```diff
@@ -144,3 +137,9 @@ export const useIsRequestRequisitionDisabled = (): boolean => {
     status === RequisitionNodeStatus.Sent
   );
 };
+
+export const useSaveRequestLines = () => {
+  const { store } = useHostContext();
```
Comment (mark-prins) [inline packages/requisitions/src/RequestRequisition/api/hooks.ts:142]: can now do
```
const { storeId } = useAuthContext();
```

**Review by mark-prins — APPROVED** [review]
🎉  looking good! having a bit of trouble testing due to the api changes.. will test more fully later

File: packages/requisitions/src/RequestRequisition/api/api.ts (hunk 17 above)
Comment (joshxg) [reply packages/requisitions/src/RequestRequisition/api/api.ts:39]: Yup, thanks nice spottin!

File: packages/requisitions/src/RequestRequisition/api/hooks.ts (hunk 19 above)
Comment (joshxg) [reply packages/requisitions/src/RequestRequisition/api/hooks.ts:143]: I've added the use of auth context .. I think we have the store in local storage, auth context, host context and a cooke, now? 😁

**Pull request merged**